x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/ZeroShotClassificationProcessor.java

```diff
@@ -0,0 +1,195 @@
+        }
+    }
+
+    static class ResultProcessor implements NlpTask.ResultProcessor {
+        private final int entailmentPos;
+        private final int contraPos;
+        private final String[] labels;
+        private final boolean isMultiLabel;
+
+        ResultProcessor(int entailmentPos, int contraPos, String[] labels, boolean isMultiLabel) {
+            this.entailmentPos = entailmentPos;
+            this.contraPos = contraPos;
+            this.labels = labels;
+            this.isMultiLabel = isMultiLabel;
+        }
+
+        @Override
+        public InferenceResults processResult(TokenizationResult tokenization, PyTorchResult pyTorchResult) {
+            if (pyTorchResult.getInferenceResult().length < 1) {
+                return new WarningInferenceResults("Zero shot classification result has no data");
+            }
+            // TODO only the first entry in the batch result is verified and
+            // checked. Implement for all in batch
+            if (pyTorchResult.getInferenceResult()[0].length != labels.length) {
+                return new WarningInferenceResults(
+                    "Expected exactly [{}] values in zero shot classification result; got [{}]",
+                    labels.length,
+                    pyTorchResult.getInferenceResult().length
+                );
+            }
+            final double[] normalizedScores;
+            if (isMultiLabel) {
+                normalizedScores = new double[pyTorchResult.getInferenceResult()[0].length];
+                int v = 0;
+                for (double[] vals : pyTorchResult.getInferenceResult()[0]) {
+                    if (vals.length != 3) {
+                        return new WarningInferenceResults(
+                            "Expected exactly [{}] values in inner zero shot classification result; got [{}]",
+                            3,
+                            vals.length
+                        );
+                    }
+                    // assume entailment is `0`, softmax between entailment and contradiction
+                    normalizedScores[v++] = NlpHelpers.convertToProbabilitiesBySoftMax(
+                        new double[]{vals[entailmentPos], vals[contraPos]}
+                    )[0];
+                }
+            } else {
+                double[] entailmentScores = new double[pyTorchResult.getInferenceResult()[0].length];
+                int v = 0;
+                for (double[] vals : pyTorchResult.getInferenceResult()[0]) {
+                    if (vals.length != 3) {
+                        return new WarningInferenceResults(
+                            "Expected exactly [{}] values in inner zero shot classification result; got [{}]",
+                            3,
+                            vals.length
+                        );
+                    }
+                    // assumes entailment is `0`, it COULD be `2`
```

Comment:
is this comment out of date? It seems to be referring to a hard-coded vale rather than the use of `entailmentPos`